yt_dlp/extractor/txxx.py

```diff
@@ -0,0 +1,307 @@
+import base64
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    parse_duration,
+    qualities,
+)
+
+
+class TxxxBaseIE(InfoExtractor):
+    # some non-standard characters are used in the base64 string
+    _BASE64_CHAR_REPL_MAP = {
+        "\u0405": "S",
+        "\u0406": "I",
+        "\u0408": "J",
+        "\u0410": "A",
+        "\u0412": "B",
+        "\u0415": "E",
+        "\u041a": "K",
+        "\u041c": "M",
+        "\u041d": "H",
+        "\u041e": "O",
+        "\u0420": "P",
+        "\u0421": "C",
+        "\u0425": "X",
+        ",": "/",
+        ".": "+",
+        "~": "=",
+    }
+
+    def _decode_base64(self, text):
+        for from_char, to_char in self._BASE64_CHAR_REPL_MAP.items():
+            text = text.replace(from_char, to_char)
+        return base64.b64decode(text).decode('utf-8')
```

Comment:
why not just use `base64.b64decode(text, altchars=...)`

Reply:
According to [this](https://docs.python.org/3/library/base64.html):
> Optional altchars must be a [bytes-like object](https://docs.python.org/3/glossary.html#term-bytes-like-object) or ASCII string of at least length 2 (additional characters are ignored) which specifies the alternative alphabet used instead of the + and / characters.

It sounds to me altchars only replaces + and / instead of an arbitrary set of characters.